plugins/c8y_firmware_plugin/src/tests.rs

```diff
@@ -0,0 +1,187 @@
+use crate::FirmwareManager;
+use c8y_api::http_proxy::MockC8YHttpProxy;
+use mockall::predicate;
+use serde_json::json;
+use sha256::digest;
+use sha256::try_digest;
+use std::sync::Arc;
+use tedge_api::OperationStatus;
+use tedge_test_utils::fs::TempTedgeDir;
+
+const TEST_TIMEOUT_MS: Duration = Duration::from_millis(5000);
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+#[serial_test::serial]
+async fn test_handle_firmware_update_request_child_device() -> anyhow::Result<()> {
+    let child_device_id = "child-device";
+    let firmware_name = "fw-name";
+    let firmware_version = "fw-version";
+
+    // Create required directories
+    let tmp_dir = TempTedgeDir::new();
+    tmp_dir.dir("cache");
+    tmp_dir.dir("file-transfer");
+    tmp_dir.dir("firmware");
+
+    let broker = mqtt_tests::test_mqtt_broker();
+    let mut c8y_http_client = MockC8YHttpProxy::new();
+
+    c8y_http_client
+        .expect_download_file()
+        .with(
+            predicate::always(),
+            predicate::always(),
+            predicate::always(),
+        )
+        .returning(|_, file_name, tmp_dir_path| {
+            let downloaded_path = tmp_dir_path.join(file_name);
+            std::fs::File::create(&downloaded_path)?;
+            Ok(downloaded_path)
+        });
+
+    let mut firmware_manager = FirmwareManager::new(
+        "tedge_device_id".to_string(),
+        broker.port,
+        Arc::new(Mutex::new(c8y_http_client)),
+        mockito::server_address().to_string(),
+        tmp_dir.to_path_buf(),
+        tmp_dir.to_path_buf(),
+        Duration::from_secs(3600),
+    )
+    .await?;
+
+    // Run the plugin's runtime logic in an async task
+    tokio::spawn(async move {
+        let _ = firmware_manager.run().await;
+    });
+
+    // Mock download endpoint for the plugin to download config file update from the cloud
+    let cloud_url_path = "/some/cloud/url";
+    let mock_http_server_host = mockito::server_url();
+    let cloud_firmware_url = format!("{mock_http_server_host}{cloud_url_path}");
+    let file_cache_key = digest(cloud_firmware_url.clone());
+    let firmware_file_path_in_cache_dir = tmp_dir
+        .path()
+        .join("firmware")
+        .join("cache")
+        .join(file_cache_key.clone());
+
+    let mut tedge_command_messages = broker
+        .messages_published_on(&format!(
+            "tedge/{child_device_id}/commands/req/firmware_update"
+        ))
+        .await;
+
+    // Send a c8y_Firmware operation to the plugin
+    broker
+        .publish(
+            "c8y/s/ds",
+            format!(
+                "515,{child_device_id},{firmware_name},{firmware_version},{cloud_firmware_url}"
+            )
+            .as_str(),
+        )
+        .await?;
+
+    // Since sha256 is generated by the downloaded file, can create RequestPayload after publishing 515.
+    // TODO! How to ignore the operation ID from the test result? It's randomly generated...
```

Comment:
We can do [partial JSON matching](https://docs.rs/assert-json-diff/latest/assert_json_diff/#partial-matching) which is supported by `assert_json_diff` used elsewhere in this project